use_cases/eluc/app/components/filter.py

```diff
@@ -0,0 +1,242 @@
+                                                    id={"type": "preset-button", "index": "low-change"},
+                                                    color="danger",
+                                                    className="me-1"
+                                                )
+                                            ]
+                                        )
+                                    ],
+                                    width={"offset": 3, "size": 3}
+                                ),
+                                dbc.Col([
+                                    self.create_outcome_sliders(),
+                                ], width=4)
+                            ]
+                        ),
+                        dcc.Store(id="results-store"),
+                        dcc.Graph(
+                            id="prescriptions"
+                        )
+                    ]
+                )
+            ]
+        )
+
+    def register_callbacks(self, app):
+        """
+        Registers callbacks for the filtering.
+        """
+        @app.callback(
+            [Output({"type": "filter-slider", "index": ALL}, param, allow_duplicate=True)
+             for param in self.slider_params],
+            Input("results-store", "data"),
+        )
+        def update_filter_sliders(results_json: dict[str: list]) -> list:
+            """
+            When the results store is updated when context is selected, we update the filter sliders to reflect the
+            min and max values of each outcomes so the user can filter by them.
+            """
+            results_df = pd.DataFrame(results_json)
+            outputs = [[] for _ in range(len(self.slider_params))]
+            for outcome in constants.CAO_MAPPING["outcomes"]:
+                min_val = results_df[outcome].min()
+                max_val = results_df[outcome].max()
+                # We need to round down for the min value and round up for the max value
+                val_range = max_val - min_val
+                min_val_rounded = min_val - val_range / 100
+                max_val_rounded = max_val + val_range / 100
+                outputs[0].append(min_val_rounded)
+                outputs[1].append(max_val_rounded)
+                outputs[2].append([min_val_rounded, max_val_rounded])
+                outputs[3].append({min_val_rounded: f"{int(100*min_val_rounded)}",
+                                   max_val_rounded: f"{int(100*max_val_rounded)}"})
+            return outputs
+
+        @app.callback(
+            Output({"type": "filter-slider", "index": ALL}, "value", allow_duplicate=True),
+            Input({"type": "preset-button", "index": ALL}, "n_clicks"),
+            State("results-store", "data"),
+            prevent_inital_call=True
+        )
+        def select_preset(n_clicks, results_json):
+            """
+            Selects a preset for the filter sliders from the 3 presets.
+            """
```

Comment:
We make the presets just slice the outcomes into thirds and each third is our preset